contracts/core/IBCClient.sol

```diff
@@ -26,43 +27,89 @@ library IBCClient {
     /**
      * @dev updateClient updates the consensus state and the state root from a provided header
      */
-    function updateClient(IBCHost host, IBCMsgs.MsgUpdateClient calldata msg_) external {
+    function updateClient(IBCHost host, IBCMsgs.MsgUpdateClient calldata msg_)
+        external
+    {
         host.onlyIBCModule();
         bytes memory clientStateBytes;
-        bytes memory consensusStateBytes;
-        Height.Data memory height;
+        ConsensusUpdate.Result[] memory results;
         bool found;
-    
+
         (clientStateBytes, found) = host.getClientState(msg_.clientId);
         require(found, "clientState not found");
 
-        (clientStateBytes, consensusStateBytes, height) = getClient(host, msg_.clientId).checkHeaderAndUpdateState(host, msg_.clientId, clientStateBytes, msg_.header);
-    
+        bool verified = getClient(host, msg_.clientId).verifyClientMessage(
+            host,
+            msg_.clientId,
+            clientStateBytes,
+            msg_.header
+        );
+
+        if (!verified) {
+            require(verified, "header verification failed");
+        }
+
+        (clientStateBytes, results) = getClient(host, msg_.clientId)
+            .updateState(host, msg_.clientId, clientStateBytes, msg_.header);
+
         //// persist states ////
         host.setClientState(msg_.clientId, clientStateBytes);
```

Comment:
@nadeemb53  we can still work with @bluele 's suggestion, instead of having the update_state method return a `ConsensusUpdateResult`,
let's do all the state updates inside `update_state` , so that way each light client can store the consensus state however they want.

So you just have to move the calls to `setClientState`, `setConsensusState`, `setProcessedTime`, `setProcessedHeight`, into `update_state` for the beefy

Reply:
For now add some documentation to the `update_state` function that it should do all the state persisting internally and return an array of consensus heights.

Reply:
Addressed this in the recent commit